src/dotnet/commands/dotnet-test/VSTestArgumentConverter.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.Cli
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// Converts the given arguments to vstest parsable arguments
+    /// </summary>
+    public class VSTestArgumentConverter
+    {
+        private const string verbosityString = "--logger:console;verbosity=";
+
+        private readonly Dictionary<string, string> ArgumentMapping = new Dictionary<string, string>
+        {
+            ["-h"] = "--help",
+            ["-s"] = "--settings",
+            ["-t"] = "--listtests",
+            ["-a"] = "--testadapterpath",
+            ["-l"] = "--logger",
+            ["-f"] = "--framework",
+            ["-d"] = "--diag",
+            ["--filter"] = "--testcasefilter",
+            ["--list-tests"] = "--listtests",
+            ["--test-adapter-path"] = "--testadapterpath",
+            ["--results-directory"] = "--resultsdirectory"
+        };
+
+        private readonly Dictionary<string, string> VerbosityMapping = new Dictionary<string, string>
+        {
+            ["q"] = "quiet",
+            ["m"] = "minimal",
+            ["n"] = "normal",
+            ["d"] = "detailed",
+            ["diag"] = "diagnostic"
+        };
+
+        private readonly string[] IgnoredArgumentArray = new string[]
+        {
+            "-c",
+            "--configuration",
+            "--runtime",
+            "-o",
+            "--output",
+            "--no-build",
+            "--no-restore",
+            "--interactive"
+        };
+
+        /// <summary>
+        /// Converts the given arguments to vstest parsable arguments
+        /// </summary>
+        /// <param name="args">original arguments</param>
+        /// <param name="ignoredArgs">arguments ignored by the converter</param>
+        /// <returns>list of args which can be passsed to vstest</returns>
+        public List<string> Convert(string[] args, out List<string> ignoredArgs)
+        {
+            var newArgList = new List<string>();
+            ignoredArgs = new List<string>();
+
+            string activeArgument = null;
+            foreach (var arg in args)
+            {
+                if (arg.StartsWith("-"))
```

Comment:
do we want to handle args staring with "/"?

Reply:
No, The help of dotnet does not suggest using the / as option, I don't want to unnecessarily add that check